### ISSUE TITLE
[Bug] App unusable, keeps spamming "added to queue"

### ISSUE THREAD

**Issue by GfEW** [NeoApplications/Neo-Store#687]
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/NeoApplications/Neo-Store/issues) and [closed](https://github.com/NeoApplications/Neo-Store/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I'm on the latest version.
- [X] I'm not using a test build (alpha/beta/release-candidate).
- [X] This issue contains only one bug.

### Describe the bug

Unfortunately, I haven't managed to understand, let alone reproduce, exactly how the problem started, but it is so severe I've had to resort to a different app, only checking NeoStore once a month for a fix (so far, to no avail).

My app collection from alternative stores (mostly f-droid) entails a few dozen apps. The major upgrade to 1.0.x has made updating with NeoStore very slow on all devices. The process got so time consuming I often had to stop watching for the next app's update dialog to finally appear. Over time, more and more stalled/incomplete updates have piled up.

Now, whenever I open NeoStore, it takes a moment to settle, then starts to issue a toast message reading _"added to queue"_ every few seconds. That message is repeated virtually endlessly, even after many hours, without any change or decrease in frequency. I can't work in other apps meanwhile because those messages keep stealing focus.

The only way I know to stop them (aside from rebooting) is to kill NeoStore. Then the next time NeoStore is launched, __one__ update dialog actually appears, and that app is updated successfully. After that, the _"added to queue"_ continues.

### Expected Behavior

For every updatable app, NeoStore should fetch the update in reasonable time and should add it to App Manager's queue only once.

### Neo Store's Version

1.0.5

### Installation Source

Official F-Droid repo

### Last Known Working Version

not sure, but never had this issue with pre-1.0.0

### Relevant information

- Device: diverse
- Android Version: 9 to 11
- ROM: (AOSP, CalyxOS, MIUI, GOS...): stock
- Disabled battery optimization for Neo Store?: yes
- What is your selected installer?: App Manager

**Comment by machiav3lli** [comment]
I'm not sure about the slowness (it could be the parallel downloads or that the servers are throatelling certain agents - or both?).

I'm aware of memory issues in 1.0.X and these were already fixed in the previous 1.0.6 alphas. I'm currently testing the beta which could turn up to be an rc (or even stable release), if test builds  pass the community.

On the AM installer issue: I'm aware of an issue with the AM installer logic, but lack any documentation how to handle this better considering lack of full documentation (or maybe an API?). I'd suggest using the default session installer or root generally. And when there's a chance (change of situation) I'll approach the AM installer issue again.

**Comment by GfEW** [comment]
Thanks for your thoughts and comments.

> I'm not sure about the slowness (it could be the parallel downloads or that the servers are throatelling certain agents - or both?).

I have no idea, other than it didn't happen with e. g. 0.9.15, and doesn't with the F-Droid app, either.

> I'm aware of memory issues in 1.0.X and these were already fixed in the previous 1.0.6 alphas. I'm currently testing the beta which could turn up to be an rc (or even stable release), if test builds pass the community.

I'm looking forward to it.

> On the AM installer issue: I'm aware of an issue with the AM installer logic, but lack any documentation how to handle this better considering lack of full documentation (or maybe an API?). I'd suggest using the default session installer or root generally. And when there's a chance (change of situation) I'll approach the AM installer issue again.

I've given the default session installer a try (no root available ATM). The setting doesn't seem to immediately stop the messages, but once NeoStore has been force-stopped and relaunched, it refrains from resuming them. Working in other apps while waiting for updates is possible again, what a relief.

The update progress (if any) is still unbearably slow, though. One single update dialog (by the session installer) has shown up shortly after relaunching NeoStore and tapping `Install all`, but it's been 3 hours since and I'm still waiting for the next update dialog (out of 35) to appear. There's no error message nor any other indication of what's going wrong.

**Comment by PowerUser64** [comment]
Is it possible that [v1.1.0](https://github.com/NeoApplications/Neo-Store/releases/tag/1.1.0) solved this for you? I had some similar performance issues that made neostore basically impossible to use, but they've been solved since v1.1.0.

**Comment by GfEW** [comment]
> Is it possible that [v1.1.0](https://github.com/NeoApplications/Neo-Store/releases/tag/1.1.0) solved this for you? I had some similar performance issues that made neostore basically impossible to use, but they've been solved since v1.1.0.

Indeed, my brief check of v1.1.0 indicates that some of the most prohibitive, long standing speed issues introduced by v1.0.0 got finally resolved. I've yet to test this particular issue though in more depth, but more urgently, examine and report an unrelated, severe regression first (that, alas, renders NS still unusable as a daily runner for me).
Will provide more details re "resolved: yes/no?" when time allows.
